### PR TITLE
PR 3 - Clean Architecture

### PR DESCRIPTION
**Pull request by AndresConforti**
## Description
PR 3 - Modified the project structure, based on the Clean Architecture.

## Package Structure
| Structure |
| ------------- |
| arch |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,14 @@
-# TheMovieDB App Part One
+# TheMovieDB App Part Three
```
Comment (matiasilveiro-gl) [inline README.md]: Good idea to update the `README.md`!! 🚀  Buuut I wouldn't delete the previous "Parts". Create new sections in the `README` with the new tasks, but without deleting the previous work :D

File: lib/src/data/repository/genre_repo.dart [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
-import './interface/genre_interface.dart';
+import '../../domain/entity/genre_data.dart';
+import '../../core/utils/strings.dart';
+import '../../domain/repository/interface/genre_interface.dart';
 
 class GenreRepository extends GenreInterface {
   List<Genre> parsedGenresList = [];
 
-  Future<List<Genre>> parseGenres() async {
+  static Future<List<Genre>> parseGenres() async {
```
Comment (matiasilveiro-gl) [inline lib/src/data/repository/genre_repo.dart]: You don't need this method to be static. Also I'd mark this as private

File: lib/src/data/repository/movie_repo.dart [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-import './interface/movie_interface.dart';
+import '../../core/utils/strings.dart';
+import '../../domain/entity/movie_data.dart';
+import '../../domain/repository/interface/movie_interface.dart';
 
 class MovieRepository extends MovieInterface {
   List<Movie> parsedMovieList = [];
 
-  Future<List<Movie>> parseMovies() async {
+  static Future<List<Movie>> parseMovies() async {
```
Comment (matiasilveiro-gl) [inline lib/src/data/repository/movie_repo.dart]: You don't need this method to be static. Also I'd mark this as private

File: lib/src/domain/entity/genre_data.dart
Comment (matiasilveiro-gl) [inline lib/src/domain/entity/genre_data.dart:1]: To follow a better clean arch, you could have this `Genre` entity in domain, and a `GenreModel extends Genre` (with the .fromJson constructor) in data.

File: lib/src/domain/entity/movie_data.dart
Comment (matiasilveiro-gl) [inline lib/src/domain/entity/movie_data.dart:1]: To follow a better clean arch, you could have this `Movie` entity in domain, and a `MovieModel extends Movie` (with the .fromJson constructor) in data.

File: lib/src/data/repository/movie_repo.dart (hunk 3 above)
Comment (AndresConforti) [reply lib/src/data/repository/movie_repo.dart]: Done.

File: lib/src/data/repository/genre_repo.dart (hunk 2 above)
Comment (AndresConforti) [reply lib/src/data/repository/genre_repo.dart]: Done.

File: README.md (hunk 1 above)
Comment (AndresConforti) [reply README.md]: Done.

File: lib/src/domain/entity/movie_data.dart
Comment (AndresConforti) [reply lib/src/domain/entity/movie_data.dart:1]: Done.

File: lib/src/domain/entity/genre_data.dart
Comment (AndresConforti) [reply lib/src/domain/entity/genre_data.dart:1]: Done.

**Review by matiasilveiro-gl — APPROVED** [review]
LGTM! 🚀

**Review by FranciscoGaraventa — APPROVED** [review]

**Pull request merged**